avm/res/app/session-pool/main.bicep

```
@@ -0,0 +1,272 @@
+  }
+  tags: tags
+}
+
+resource sessionPool_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock ?? {}) && lock.?kind != 'None') {
+  name: lock.?name ?? 'lock-${name}'
+  properties: {
+    level: lock.?kind ?? ''
+    notes: lock.?kind == 'CanNotDelete'
+      ? 'Cannot delete resource or child resources.'
+      : 'Cannot delete or modify the resource or child resources.'
+  }
+  scope: sessionPool
+}
+
+resource sessionPool_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-04-01' = [
+  for (roleAssignment, index) in (formattedRoleAssignments ?? []): {
+    name: roleAssignment.?name ?? guid(sessionPool.id, roleAssignment.principalId, roleAssignment.roleDefinitionId)
+    properties: {
+      roleDefinitionId: roleAssignment.roleDefinitionId
+      principalId: roleAssignment.principalId
+      description: roleAssignment.?description
+      principalType: roleAssignment.?principalType
+      condition: roleAssignment.?condition
+      conditionVersion: !empty(roleAssignment.?condition) ? (roleAssignment.?conditionVersion ?? '2.0') : null // Must only be set if condtion is set
+      delegatedManagedIdentityResourceId: roleAssignment.?delegatedManagedIdentityResourceId
+    }
+    scope: sessionPool
+  }
+]
+
+@description('The name of the session pool.')
+output name string = sessionPool.name
+@description('The resource ID of the deployed session pool.')
+output resourceId string = sessionPool.id
+
+@description('The name of the resource group in which the session pool was created.')
+output resourceGroupName string = resourceGroup().name
+@description('The management endpoint of the session pool.')
+output managementEndpoint string = sessionPool.properties.poolManagementEndpoint
+
+@description('The principal ID of the system assigned identity.')
+output systemAssignedMIPrincipalId string = sessionPool.?identity.?principalId ?? ''
+
+@description('Optional. Custom container definition.')
+type sessionContainerType = {
+  @description('Optional. Container start command arguments.')
+  args: string[]?
+
+  @description('Optional. Container start command.')
+  command: string[]?
+
+  @description('Optional. Container environment variables.')
+  env: environmentVarType?
+
+  @description('Required. Container image tag.')
+  image: string
+
+  @description('Required. Custom container name.')
+  name: string
+
+  @description('Required. Container resource requirements.')
+  resources: sessionContainerResourcesType
+}[]?
+
+@description('Optional. Environment variable definition for a container. Only used with custom containers.')
+type environmentVarType = {
+  @description('Required. Environment variable name.')
+  name: string
+
+  @description('Optional. Required if value is not set. Name of the Container App secret from which to pull the environment variable value.')
+  secretRef: string?
+
+  @description('Optional. Required if secretRef is not set. Non-secret environment variable value.')
+  value: string?
+}[]?
```

All types should be non nullable and singular
```suggestion
}
```

---

Apologies for not catching these the first time around.  Done!